In [2]:
# Importing packages 
import numpy as np
import pandas as pd
import sys
import math
import matplotlib.pyplot as plt

**Part 1**

Determine the financial performance of the organization from the Urban Ministries of Durham the first digit of the net income, expenses and income-expenses will be considered. Theoretically, this should follow a Bendford distribution and any deviations from this distribution would suggest fabricated numbers. This part of the project conducts the statistical analysis and R-studio will create the graph.

In [ ]:
# Import Data Tax Data 
df = pd.read_csv(r'C:\Users\15056\Documents\BIOS611\bios611-projects-fall-2019-jvazquez2\project_3\data\IRS.csv')
df = df[df.Net >= 0]
data = df['Net']

The below code will count the first digit of each number. This part of the project was 
facilitated by Elena C. You can find the original code in her Github page: https://github.com/eleprocha/Benford-s-Law_python_code/blob/master/code Additional modifications of the code can be found below.

In [36]:
# Counting the first digit
def count_first_digit(data_str):
    mask=df[data_str]>1.
    data=list(df[mask][data_str])
    for i in range(len(data)):
        while data[i]>10:
            data[i]=data[i]/10
    first_digits=[int(x) for x in sorted(data)]
    unique=(set(first_digits))#a list with unique values of first_digit list
    data_count=[]
    for i in unique:
        count=first_digits.count(i)
        data_count.append(count)
    total_count=sum(data_count)
    data_percentage=[(i/total_count)*100 for i in data_count]
    return  total_count,data_count, data_percentage

total_count, data_count_pre, data_percentage_pre = count_first_digit("Net")

# In this case data_count_pre and data_percentage_pre have the format of [1,3,4,5,6,8]. 
# Need to add zeros for 2, 7, 9 
data_count = []
data_percentage = []
j = 0
for i in range(1,10):
        if i in (2,7,9):
            data_count.append(0)
            data_percentage.append(0)
        else:
            data_count.append(data_count_pre[j])
            data_percentage.append(data_percentage_pre[j])
            j = j+1
            
            
# Benford's Law percentages for leading digits 1-9
BENFORD = [30.1, 17.6, 12.5, 9.7, 7.9, 6.7, 5.8, 5.1, 4.6]

In [44]:
# Get the expected number of counts
def get_expected_counts(total_count):
    return [round(p * total_count / 100) for p in BENFORD]

expected_counts=get_expected_counts(total_count)

# Observed vs Expected  
chi_square_stat = 0  # chi square test statistic
for data, expected in zip(data_count,expected_counts):
    chi_square = math.pow(data - expected, 2)
    chi_square_stat += chi_square / expected

print("\nChi-squared Test Statistic = {:.3f}".format(chi_square_stat))
print("Critical value at a P-value of 0.05 is 15.51.")


Chi-squared Test Statistic = 9.500
Critical value at a P-value of 0.05 is 15.51.


Since our chi-square test statistic of 9.50 was less than the critical calue of 15.15 we fail to reject the null hypothesis. We conclude that the income tax data from Urban Ministries of Durham does not provide sufficient evidence to claim that the non-for-profit engages in fraudulent behavior when reporting taxes. One of the the limitations from this analysis was that we had few observations. One of the recommendations for the test is to have at least 50 obs but in this case we had only 12. 

**Part 2**

Count the number of patients that came in with a condition (e.g. - mental health problem) and left with (1) none, (2) the same, or (3) with more problems. The change in health related conditions will be illustrated using bar-plots, stratified by ethnicity in R-studio.

In [48]:
# Import Client Data
my_client = pd.read_csv("https://raw.githubusercontent.com/biodatascience/datasci611/gh-pages/data/project2_2019/CLIENT_191102.tsv", sep = '\t')
my_disability_entry = pd.read_csv("https://raw.githubusercontent.com/biodatascience/datasci611/gh-pages/data/project2_2019/DISABILITY_ENTRY_191102.tsv", sep = '\t')
my_disability_exit = pd.read_csv("https://raw.githubusercontent.com/biodatascience/datasci611/gh-pages/data/project2_2019/DISABILITY_EXIT_191102.tsv", sep = '\t')

EE Provider ID  EE UID  \
0  Urban Ministries of Durham - Durham County - S...  687901   
1  Urban Ministries of Durham - Durham County - S...  687901   
2  Urban Ministries of Durham - Durham County - S...  687901   
3  Urban Ministries of Durham - Durham County - S...  687901   
4  Urban Ministries of Durham - Durham County - S...  687901   

       Client Unique ID  Client ID Disability Determination (Entry)  \
0  pbkf09291954p610b236     397941                         No (HUD)   
1  pbkf09291954p610b236     397941                         No (HUD)   
2  pbkf09291954p610b236     397941                         No (HUD)   
3  pbkf09291954p610b236     397941                         No (HUD)   
4  pbkf09291954p610b236     397941                         No (HUD)   

             Disability Type (Entry) Disability Start Date (Entry)  \
0                Alcohol Abuse (HUD)                     4/20/2015   
1  Both Alcohol and Drug Abuse (HUD)                     4/20/2015   
2     Chronic Health Condition (HUD)                     4/20/2015   
3                Developmental (HUD)                     4/20/2015   
4                   Drug Abuse (HUD)                     4/20/2015   

  Disability End Date (Entry)  \
0                         NaN   
1                         NaN   
2                         NaN   
3                         NaN   
4                         NaN   

                            Provider (417-provider)  \
0  Urban Ministries of Durham - Durham County(1562)   
1  Urban Ministries of Durham - Durham County(1562)   
2  Urban Ministries of Durham - Durham County(1562)   
3  Urban Ministries of Durham - Durham County(1562)   
4  Urban Ministries of Durham - Durham County(1562)   

   Recordset ID (417-recordset_id) Date Added (417-date_added)  
0                          2261529                   7/16/2015  
1                          2261530                   7/16/2015  
2                          2261524                   7/16/2015  
3                          2261527                   7/16/2015  
4                          2261528                   7/16/2015

In [53]:
client_data = pd.merge(my_disability_entry, my_disability_exit, on = 'Client ID', how = 'right')
client_data.head()
mydisability_exit.head()

NameError: name 'mydisability_exit' is not defined

In [55]:
my_disability_exit.head()

EE Provider ID  EE UID  \
0  Urban Ministries of Durham - Durham County - S...  687901   
1  Urban Ministries of Durham - Durham County - S...  687901   
2  Urban Ministries of Durham - Durham County - S...  687901   
3  Urban Ministries of Durham - Durham County - S...  687901   
4  Urban Ministries of Durham - Durham County - S...  687901   

       Client Unique ID  Client ID Disability Determination (Exit)  \
0  pbkf09291954p610b236     397941                        No (HUD)   
1  pbkf09291954p610b236     397941                        No (HUD)   
2  pbkf09291954p610b236     397941                        No (HUD)   
3  pbkf09291954p610b236     397941                        No (HUD)   
4  pbkf09291954p610b236     397941                        No (HUD)   

              Disability Type (Exit) Disability Start Date (Exit)  \
0                Alcohol Abuse (HUD)                    4/20/2015   
1  Both Alcohol and Drug Abuse (HUD)                    4/20/2015   
2     Chronic Health Condition (HUD)                    4/20/2015   
3                Developmental (HUD)                    4/20/2015   
4                   Drug Abuse (HUD)                    4/20/2015   

  Disability End Date (Exit)  \
0                        NaN   
1                        NaN   
2                        NaN   
3                        NaN   
4                        NaN   

                            Provider (417-provider)  \
0  Urban Ministries of Durham - Durham County(1562)   
1  Urban Ministries of Durham - Durham County(1562)   
2  Urban Ministries of Durham - Durham County(1562)   
3  Urban Ministries of Durham - Durham County(1562)   
4  Urban Ministries of Durham - Durham County(1562)   

   Recordset ID (417-recordset_id) Date Added (417-date_added)  
0                          2261529                   7/16/2015  
1                          2261530                   7/16/2015  
2                          2261524                   7/16/2015  
3                          2261527                   7/16/2015  
4                          2261528                   7/16/2015